e2e/recipes/recipes.go

```diff
@@ -0,0 +1,210 @@
+
+	defaultGroupName = "rg1"
+	backup           = "backup"
+	restore          = "restore"
+	group            = "group"
+)
+
+// supportedSelectResources keeps supported select resources.
+var supportedSelectResources = map[string]struct{}{
+	"deployment":  {},
+	"pod":         {},
+	"statefulset": {},
+}
+
+func Generate(ctx types.TestContext, recipeConfig *config.Recipe) (*recipe.Recipe, error) {
+	ctxName := ctx.Name()
+	appNamespace := ctx.AppNamespace()
+
+	hooks, err := generateHooks(ctx, recipeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate hooks: %w", err)
+	}
+
+	return &recipe.Recipe{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       kindRecipe,
+			APIVersion: recipeAPIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ctxName,
+			Namespace: appNamespace,
+		},
+		Spec: recipe.RecipeSpec{
+			Hooks:     hooks,
+			Groups:    generateGroups(ctx, appNamespace),
+			Workflows: generateWorkflows(recipeConfig, hooks),
+		},
+	}, nil
+}
+
+func generateHooks(ctx types.TestContext, rc *config.Recipe) ([]*recipe.Hook, error) {
+	var hooks []*recipe.Hook
+
+	if rc.CheckHook {
+		checkHook, err := prepareCheckHook(ctx)
+		if err != nil {
+			return hooks, err
+		}
+
+		hooks = append(hooks, checkHook)
+	}
+
+	if rc.ExecHook {
+		execHook, err := prepareExecHook(ctx)
+		if err != nil {
+			return hooks, err
+		}
+
+		hooks = append(hooks, execHook)
+	}
+
+	return hooks, nil
+}
+
+func prepareCheckHook(ctx types.TestContext) (*recipe.Hook, error) {
+	if checks := ctx.Workload().GetChecks(ctx.AppNamespace()); len(checks) == 0 {
+		return nil, fmt.Errorf("failed to prepare check hooks, no checks defined for workload %q",
+			ctx.Workload().GetName())
+	}
+
+	selectResource := ctx.Workload().GetSelectResource()
+	if _, ok := supportedSelectResources[selectResource]; !ok {
+		return nil, fmt.Errorf("selectResource %q is not supported for check hook in workload %q (choose from %v)",
+			selectResource, ctx.Workload().GetName(), maps.Keys(supportedSelectResources))
+	}
+
+	return &recipe.Hook{
+		Name:           "check-hook",
+		Type:           "check",
+		Namespace:      ctx.AppNamespace(),
+		LabelSelector:  ctx.Workload().GetLabelSelector(),
+		SelectResource: ctx.Workload().GetSelectResource(),
+		Timeout:        defaultHookTimeout,
+		Chks:           ctx.Workload().GetChecks(ctx.AppNamespace()),
+	}, nil
+}
+
+func prepareExecHook(ctx types.TestContext) (*recipe.Hook, error) {
+	if operations := ctx.Workload().GetOperations(ctx.AppNamespace()); len(operations) == 0 {
+		return nil, fmt.Errorf("failed to create exec hook, no operations defined for workload %q", ctx.Workload().GetName())
+	}
+
+	selectResource := ctx.Workload().GetSelectResource()
+	if _, ok := supportedSelectResources[selectResource]; !ok {
+		return nil, fmt.Errorf("selectResource %q is not supported for exec hook in workload %q (choose from %v)",
+			selectResource, ctx.Workload().GetName(), maps.Keys(supportedSelectResources))
+	}
```

Comment:
So we have the same code for getting and validating selectResource. Instead of repeating this code we can add a helper for getting the value and validating it, so we can do here:

    selectResource, err := validateSelectResource(ctx.Workload())
    if err != nil {
        return fmt.Errorf("failed to create exec hook: %w", err)
    }

Can be something like:

```go
func validateSelectResource(w types.Workload) (string, error) {
    sr := w.GetSelectResource()
	if _, ok := supportedSelectResources[sr]; !ok {
		return sr, fmt.Errorf("selectResource %q is not supported (choose from %v)",
			selectResource, slices.Collect(maps.Keys(sr)))
	}
    
    return sr, nil
}
```